### PR TITLE
update code comment that are out of date for apps/v1beta2

### LINKED ISSUE #54139: should update out of date code comment for apps/v1beta2
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
> /kind bug

**What happened**:

PR https://github.com/kubernetes/kubernetes/pull/50164 have remove the remove apps/v1beta2 defaulting codes for obj.Spec.Selector and obj.Labels, but the API reference and code comment is not updated and it is confused for user, releated issue https://github.com/kubernetes/kubernetes/issues/53976

**What you expected to happen**:

update code comment

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration**:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by wackxu**
**What this PR does / why we need it**:

update comment that are out of date

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #54139

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by lavalamp** [comment]
@kow3ns @janetkuo are these comment changes correct?

File: staging/src/k8s.io/api/apps/v1beta2/types.go [hunk 1]
```diff
@@ -169,9 +169,8 @@ type StatefulSetSpec struct {
 	Replicas *int32 `json:"replicas,omitempty" protobuf:"varint,1,opt,name=replicas"`
 
 	// selector is a label query over pods that should match the replica count.
-	// If empty, defaulted to labels on the pod template.
+	// Must be specified that matches the labels present on the pod template.
```
Comment (kow3ns) [inline staging/src/k8s.io/api/apps/v1beta2/types.go]: /s/"Must be specified that matches the labels present on the pod template"/"It must match the pod template's labels."

**Review by kow3ns — CHANGES_REQUESTED** [review]
@lavalamp The idea behind this PR is correct. 
@wackxu Please update the comments are grammatically correct.

File: staging/src/k8s.io/api/apps/v1beta2/types.go (hunk 1 above)
Comment (wackxu) [reply staging/src/k8s.io/api/apps/v1beta2/types.go]: @kow3ns addressed, PTAL

**Review by kow3ns — APPROVED** [review]

**Comment by wackxu** [comment]
Kindly ping @lavalamp for approval

File: staging/src/k8s.io/api/apps/v1beta2/types.go [hunk 2]
```diff
@@ -294,7 +293,7 @@ type DeploymentSpec struct {
 
 	// Label selector for pods. Existing ReplicaSets whose pods are
 	// selected by this will be the ones affected by this deployment.
-	// +optional
+	// It must match the pod template's labels.
 	Selector *metav1.LabelSelector `json:"selector,omitempty" protobuf:"bytes,2,opt,name=selector"`
```
Comment (janetkuo) [inline staging/src/k8s.io/api/apps/v1beta2/types.go]: Would you remove `omitempty` tags too? Ref
https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#optional-vs-required

**Comment by janetkuo** [comment]
We should probably cherrypick this back to 1.8

File: staging/src/k8s.io/api/apps/v1beta2/types.go [hunk 3]
```diff
@@ -707,10 +705,9 @@ type ReplicaSetSpec struct {
 	MinReadySeconds int32 `json:"minReadySeconds,omitempty" protobuf:"varint,4,opt,name=minReadySeconds"`
 
 	// Selector is a label query over pods that should match the replica count.
-	// If the selector is empty, it is defaulted to the labels present on the pod template.
 	// Label keys and values that must match in order to be controlled by this replica set.
+	// Must be specified that matches the labels present on the pod template.
```
Comment (crimsonfaith91) [inline staging/src/k8s.io/api/apps/v1beta2/types.go]: `It must match the pod template's labels.` is clear enough.

Comment (wackxu) [reply staging/src/k8s.io/api/apps/v1beta2/types.go]: @crimsonfaith91 Addressed

File: staging/src/k8s.io/api/apps/v1beta2/types.go (hunk 2 above)
Comment (wackxu) [reply staging/src/k8s.io/api/apps/v1beta2/types.go]: @janetkuo Addressed， PTAL

**Review by janetkuo — APPROVED** [review]

**Comment by wackxu** [comment]
kindly ping @smarterclayton @liggitt for approval

**Comment by wackxu** [comment]
@janetkuo fix kubeadm selfhosting test error, PTAL

**Comment by wackxu** [comment]
@janetkuo @kow3ns The selfhosting test has been resolved by other pr then remove the conflict code. PTAL

**Pull request merged**